src/Zenaton/Exceptions/ApiException.php

```diff
@@ -18,4 +18,33 @@ public static function unexpectedStatusCode($code)
     {
         return new static("The API responded with an unexpected status code: {$code}.");
     }
+
+    /**
+     * Creates a new instance of the exception when a connection error was received.
+     *
+     * @return self
+     */
+    public static function connectionError(\Exception $previous)
+    {
+        return new static("A connection error occurred while trying to send a request to the Zenaton API: {$previous->getMessage()}.", 0, $previous);
+    }
+
+    /**
+     * Creates a new instance of the exception when an error list is received.
+     *
+     * @return self
+     */
+    public static function fromErrorList(array $errors)
+    {
+        $errorMessages = implode("\n", array_map(static function ($error) {
+            return '  - '.$error->message;
```

Comment:
I'm not sure it could be that useful because users do not have any control on GraphQL requests made. So the path is not really their problem.
But I'll keep that in mind to see if we can use it some way, that's a good suggestion 👍